src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/DiagnosticInfo.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Interop
+{
+    public record DiagnosticInfo
+    {
+        public required DiagnosticDescriptor Descriptor { get; init; }
+        public required SequenceEqualImmutableArray<string> MessageArgs { get; init; }
+        public required Location? Location { get; init; }
+        public required IEnumerable<Location>? AdditionalLocations { get; init; }
+        public required ValueEqualityImmutableDictionary<string, string>? Properties { get; init; }
+
+        public Diagnostic ToDiagnostic() => Diagnostic.Create(
+            Descriptor,
+            Location,
+            additionalLocations: AdditionalLocations,
+            properties: Properties?.Map,
+            messageArgs: MessageArgs.Array.ToArray());
+
+        public static DiagnosticInfo Create(DiagnosticDescriptor descriptor, CodeAnalysis.Location location, params object?[] messageArgs)
```

Comment:
Why do we need to qualify the `Location` type as `CodeAnalysis.Location`. What's the conflicting type?

Reply:
Oh that was left over from some auto refactoring / renaming with the LocationInfo type, which I realized we don't actually need and removed.